### ISSUE TITLE
Feature request: Support tag+digest

### ISSUE THREAD

**Issue by rarkins** [containers/buildah#3331]
I would like if `buildah` can support having both tag+digest in the `FROM` command, like `docker` supports. Example:

```
FROM ubuntu:20.04@sha256:aba80b77e27148d99c034a987e7da3a287ed455390352663418c0f2ed40417fe
```

In scenarios like this, the tag is ignored and the digest is authoritative. Digests are important for security and deterministic build reasons, but there are benefits in including the tag too:

1. For humans, it's good information to know that it's Ubuntu 20.04 and not some random digest
2. For automation bots like [Renovate Bot](https://github.com/renovatebot/renovate) it has become a convention to mean "keep this digest automatically updated to this tag".

**Comment by rarkins** [comment]
Previous discussion: #1407

Implementation of this is already supported by CRI-O: https://github.com/cri-o/cri-o/issues/2351

If there's no opposition to this now, and if so can anyone familiar with the codebase provide some pointers on where the implementation would need to be changed?

**Comment by vrothberg** [comment]
Thanks for reaching out, @rarkins!

Good news since this is supported now.
```
buildah (master) $ ./bin/buildah  bud -f F
STEP 1/2: FROM docker.io/library/alpine:latest@sha256:234cb88d3020898631af0ccbbcca9a66ae7306ecd30c9720690858c1b007d2a0
STEP 2/2: RUN echo "It works!"
It works!
COMMIT
Getting image source signatures
Copying blob 72e830a4dff5 skipped: already exists  
Copying blob 414773de57a5 done  
Copying config 9027a75246 done  
Writing manifest to image destination
Storing signatures
--> 9027a752466
9027a75246676d0d5e5449836bf10fe6f75a4379eb96cf02bc80d52af04a541b
```

**Comment by rarkins** [comment]
@vrothberg thanks, that's great news! Is that already in the latest released/stable `buildah`?

**Comment by vrothberg** [comment]
> @vrothberg thanks, that's great news! Is that already in the latest released/stable `buildah`?

Unfortunately not but it will be released with Buildah v1.22.0.
